Feature/n2 158 create csv for videos

Dimithri opened the pull request:
JIRA Ticket: https://ackama.atlassian.net/browse/N2-158

This creates a download for the video file details which are public as a csv. Link to download the file has been added as a link after the gloss csv download link in manage lexicon page.

![image](https://user-images.githubusercontent.com/5234605/179647993-dd55d92b-e4be-497e-b128-6fcfc327da35.png)

Dimithri committed:
Initial commit with working download for video details

Dimithri committed:
Making csv creation faster

jonholdsworth commented on signbank/dictionary/adminviews.py:
```diff
@@ -713,6 +713,7 @@ def get_video_details(request):
     writer = csv.writer(response)
 
     csv_queryset =GlossVideo.objects.filter(is_public=True)
+    # return render_to_csv_response(csv_queryset)
```
Are you deliberately leaving this comment?

Dimithri replied:
No, I'm not. Must have slipped out when I was removing other comments. Will remove it :smile:

jonholdsworth reviewed (approved):
Nice work!

Dimithri committed:
Removing unnecessary comment

joshmcarthur commented on signbank/dictionary/adminviews.py:
```diff
@@ -698,6 +698,48 @@ def serialize_glosses_csv(dataset, queryset):
+
+            # If the value contains ';', put it in quotes.
+            if value and ";" in value:
+                row.append('"{}"'.format(value))
```
Does Python not handle escaping characters for you?

jonholdsworth replied:
Sorry to butt in, but - is it safe to escape characters in a CSV formatted file? Will they not appear with the `'\' `or whatever in the target output?  Or does that not matter in this case?

joshmcarthur reviewed (changes requested):
I think that there's a vital piece of information here - the video URL 😆

Dimithri committed:
Making csv creation faster

joshmcarthur committed:
Move glossvideo export into the video application instead of the primary dictionary application

joshmcarthur committed:
Add support for generating a public URL to an asset, assuming the object already has public-read ACL.

joshmcarthur committed:
Implement GlossVideo#set_public to manage the ACL of glossvideo assets on S3.

This change updates assets on S3 to be publicly readable via their URL when the glossvideo is published. If it is hidden, the asset is marked as private again.

joshmcarthur committed:
Add missing import for adminviews

joshmcarthur commented:
@Dimithri, @jonholdsworth I'll leave this up to you to review, but I've extended the glossvideo model to support manging the access-control-level of the object in S3 when the glossvideo record is published or unpublished. When the record is published, it will be marked as ACL `public-read`, meaning that no signed URL is needed to access the object. When the record is unpublished, it is marked as `private`, meaning that a signed URL is needed for access. 

This means we can export non-signed URLs in the CSV export. There isn't a way to do this with boto3, but because we know the URL format, we can just generate a string that we know is right. This means the export is super fast, under a second for the entire set of migrated assets.

Finally, I moved the export itself to the video django app, since that's where it should be. It is invoked from the same place, it just uses the video views.py definition.

jonholdsworth reviewed (approved):
I don't know enough to comment on the fine implementation details, but I understand what this change is doing. I couldn't find any obvious coding mistakes and all the dead comments have been cleaned up.
All in all this is great little piece of work folks, well done.
